backend/cmd/headlamp.go

```diff
@@ -437,6 +657,131 @@ func StartHeadlampServer(config *HeadlampConfig) {
 	log.Fatal(http.ListenAndServe(":"+config.port, handler))
 }
 
+func GetFreePort() (int, error) {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		return 0, err
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return 0, err
+	}
+
+	defer l.Close()
+
+	return l.Addr().(*net.TCPAddr).Port, nil
+}
+
+// nolint: funlen
+func (c *HeadlampConfig) startPortForward(p PortForwardPayload, token string) {
+	ports := []string{fmt.Sprintf(p.Port + ":" + p.TargetPort)}
+	ctxtProxy, ok := c.contextProxies[p.Cluster]
+
+	if !ok {
+		log.Fatalf("cluster %s not found", p.Cluster)
+		return
+	}
+
+	var caData []byte
+
+	var err error
+
+	if caData, err = ctxtProxy.context.cluster.getCAData(); err != nil {
+		log.Fatalf("error getting caData" + err.Error())
```

Comment:
same, bottle up the error and return the error in the API response.

Reply:
fixed